File: MegaHerdt.Models/Models/ArticleOffer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MegaHerdt.Models.Models
{
    public class ArticleOffer
    {
        [Key]
        public int Id { get; set; }
        [ForeignKey("Article")]
        public int ArticleId { get; set; }
        [Range(0,100)]
        public int DiscountPercentage { get; set; }
        [Required]
        public DateTime StartDate { get; set; }
        [Required]
        public DateTime EndDate { get; set; }
        public bool Enabled { get; set; } = true;
        public Article Article { get; set; }
        
    }
}

File: MegaHerdt.FrontEnd/MegaHerdt/src/app/models/ArticleOffer/ArticleOffer.ts
import { Article } from "../Article/Article";

export class ArticleOffer{
    id: number;
    discountPercentage: number;
    startDate: Date;
    endDate: Date;
    articleId: number;
    article: Article;

    constructor(id: number=0, discountPercentage: number=0, startDate: Date=new Date(), 
                 endDate: Date=new Date(), articleId: number=0, article: Article = new Article()) 
    {
        this.id = id;
        this.discountPercentage = discountPercentage;
        this.startDate = startDate;
        this.endDate = endDate;
        this.articleId = articleId;
        this.article = article;
    }
}